### PR TITLE
support for {!collapse} on `long` fields (fullstorydev/lucene-solr#187)

### PR DESCRIPTION
**Pull request by magibney**
See: https://fullstory.atlassian.net/browse/SAI-4380
backport of 8x PR: [fullstorydev/lucene-solr#187](https://github.com/fullstorydev/lucene-solr/pull/187)

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/search/TestBlockCollapse.java [hunk 1]
```diff
@@ -437,16 +443,17 @@ public void testSimple() {
...
-              // NOTE: this limitation doesn't apply to block collapse on int,
-              // so we only check 'opt.endsWith' (if ends with block hint we're ok)
-              assertQEx(
-                  "expected known limitation of using long for min/max selector when doing numeric collapse",
-                  "min/max must be Int or Float",
-                  req("q", "*:*", "fq", "{!collapse " + opt + nullPolicy + selector + "}"),
-                  SolrException.ErrorCode.BAD_REQUEST);
-              continue;
-            }
+            // if (selector.endsWith("_l") && opt.endsWith("_i")) {
```
Comment (hiteshk25) [inline solr/core/src/test/org/apache/solr/search/TestBlockCollapse.java:446]: should we remove this code?

File: solr/core/src/test/org/apache/solr/search/TestBlockCollapse.java [hunk 2]
```diff
@@ -503,16 +510,17 @@ public void testSimple() {
...
```
Comment (hiteshk25) [inline solr/core/src/test/org/apache/solr/search/TestBlockCollapse.java:513]: same here, should we remove this code?

**Comment by hiteshk25** [comment]
Merging this for now!

**Review by hiteshk25 — APPROVED** [review]
LGTM

**Comment by magibney** [comment]
Thanks for going ahead with merging this! (I think you're probably right that deleting the commented-out code is the right thing to do, btw).

**Pull request merged**